slick/src/sphinx/code/Connection.scala

```diff
@@ -103,6 +105,29 @@ object Connection extends App {
       val f: Future[Unit] = db.run(a)
       //#transaction
       Await.result(f, Duration.Inf)
+    };{
+      //#rollback
+      val insertCoffeesAction = coffees ++= Seq (
+        ("Cold_Drip", new SerialBlob(Array[Byte](101))),
+        ("Dutch_Coffee", new SerialBlob(Array[Byte](49)))
+      )
+
+      val countAction = coffees.length.result.map { coffeesCount: Int =>
+        if(coffeesCount < 5000) {
+          DBIO.failed(new Exception())
+        }
+      }
```

Comment:
Sure, I can switch to `flatMap`. The idea here was to emphasize the forced failure in `countAction`, as when the `combinedActions` runs using `transactionally`, if the `insertCoffeesAction` fails, it will roll back on its own anyway.

Just to make sure I understand, do you mean something like the below?

```scala
val insertCoffeesAction = coffees ++= Seq (
  ("Cold_Drip", new SerialBlob(Array[Byte](101))),
  ("Dutch_Coffee", new SerialBlob(Array[Byte](49)))
)

insertCoffeesAction.flatMap { _ =>
  coffees.length.result.map { coffeesCount: Int =>
    if(coffeesCount < 5000) {
      DBIO.failed(new Exception())
    } else {
      DBIO.successful()
    }
  }
}.transactionally.asTry.map {
  case Failure(_) => "Supply just can't keep up with demand."
  case Success(_) => "Now that is a lot of coffee variety."
}
```